### ISSUE TITLE
Erlang and Elixir shell lexers don't recognize node names in prompts

### ISSUE THREAD

**Issue by Anteru** [pygments/pygments-migration-test#1225]
(Original issue 1521 created by yuce on 2019-06-02T07:35:53.462037+00:00)

Erlang and Elixir shell prompts may have the node name as a prefix. `lexers.ErlangShellLexer`  doesn't highlight the prompt in the text below:

```erlang
Erlang/OTP 22 [erts-10.4] [source] [64-bit] [smp:8:8] [ds:8:8:10] [async-threads:1] [hipe]

Eshell V10.4  (abort with ^G)
(e1@example.com)1> {any, c1@example.com} ! {hello, world}.
{hello,world}
(e1@example.com)2> flush().
Shell got {hello,world}
ok
```

Updating `lexers.ErlangShellLexer` fixes the issue:

```
_prompt_re = re.compile(r'(?:\([\w@_.]+\))?\d+>(?=\s|\Z)')
```

`lexers.ElixirConsoleLexer` doesn’t highlight the prompt in the following text:

```
Erlang/OTP 22 [erts-10.4] [source] [64-bit] [smp:8:8] [ds:8:8:10] [async-threads:1] [hipe]

Interactive Elixir (1.8.1) - press Ctrl+C to exit (type h() ENTER for help)
iex(c1@ichi.cs)1> IO.puts("hello!")
hello!
:ok
```

Updating `lexers.ElixirConsoleLexer` fixes the issue:

```
_prompt_re = re.compile(r'(iex|\.{3})((?:\([\w@_.]+\))?\d+|\(\d+\))?> ')
```

**Comment by Anteru** [comment]
(Original issue 1521 last updated on 2019-06-22T19:02:50.712884+00:00)

(Issue automaticaly closed due to status in Bitbucket: resolved)

**Comment by Anteru** [comment]
(Original comment by yuce on 2019-06-02T08:21:32.731432+00:00)

PR at: [https://bitbucket.org/birkenfeld/pygments-main/pull-requests/823/fixes-1521-erlang-and-elixir-shell-lexers/diff](https://bitbucket.org/birkenfeld/pygments-main/pull-requests/823/fixes-1521-erlang-and-elixir-shell-lexers/diff)

**Comment by Anteru** [comment]
(Original comment by Anteru on 2019-06-22T19:02:49.883160+00:00)

Merged in yuce/pygments-main/fix-1521 (pull request #823)

Fixes #1521, Erlang and Elixir shell lexers

→ <<cset 7b333c00cb64>>

**Comment by Anteru** [comment]
(Original comment by Anteru on 2019-06-22T19:02:50.582509+00:00)

Fixes #1521, Erlang and Elixir shell lexers

→ <<cset 0f8d2e8ec84e>>

**Comment by Anteru** [comment]
(Original comment by Anteru on 2019-06-22T19:02:50.704371+00:00)

Merged in yuce/pygments-main/fix-1521 (pull request #823)

Fixes #1521, Erlang and Elixir shell lexers

→ <<cset 7b333c00cb64>>
